SW-3113 change text of hardware malfunctions

JFSolution opened the pull request:
change text
add new errors
solve SW-3095

Commit:
get rid of no_compressor_xxx errors
changed compressor 0 rpm error

Commit:
improve existing errors

Commit:
add fan not spinning error

Commit:
fix request_available_malfunctions

Commit:
improve freshdesk widget

Commit:
SW-3095 fix log error of camera errors

Commit:
except PiCameraValueError

Commit:
don't override existing error, update them

Commit:
add translations

Commit:
fix errors

irlaec commented on octoprint_mrbeam/__init__.py:
```diff
@@ -2319,7 +2319,7 @@ def handle_hardware_error_request(self, data):
             NO_CONTENT
         """
-        self.iobeam.send_malfunction_request()
+        self.iobeam.request_available_malfunctions()
```
Much better :)

JFSolution replied:
This was already cahnged but I missed to change the use as well 😅

irlaec commented on octoprint_mrbeam/iobeam/compressor_handler.py:
```diff
@@ -144,11 +144,10 @@ def _handle_static_data(self, payload):
-                self._hw_malfunction_handler.report_hw_malfunction_from_plugin(
-                    malfunction_id="compressor", msg="no_compressor_static_data"
-                )
+            self._logger.warn(
```
We don't want to check if there is a compressor anymore? Just to make sure

JFSolution replied:
This should be fine as we have the 0 rpm error still and the errorcodes from the selftest. I kept it as a log to see if we would need to show a error as well

irlaec commented on octoprint_mrbeam/iobeam/compressor_handler.py:
```diff
@@ -175,7 +174,9 @@ def _handle_dynamic_data(self, payload):
                         self.set_compressor_off()
                         self._hw_malfunction_handler.report_hw_malfunction_from_plugin(
-                            malfunction_id="compressor", msg="compressor_rpm_0"
+                            malfunction_id="err_compressor_malfunction",
```
Maybe it's a good idea to have all the malfunction ids, messages and error codes in the same place? Then the next time we need to change something they are easy to find. What do you think?

JFSolution replied:
They shouldn't change, but I like the idea that you can see them all in one place

irlaec commented on octoprint_mrbeam/iobeam/hw_malfunction_handler.py:
```diff
@@ -98,6 +105,8 @@ def report_hw_malfunction(self, dataset, from_plugin=False):
         for malfunction_id, data in dataset.items():
             data = data or {}
             msg = data.get("msg", malfunction_id)
+            # if malfunction_id == "compressor_malfunction":
```
Please remove this if not needed

irlaec commented on octoprint_mrbeam/iobeam/hw_malfunction_handler.py:
```diff
@@ -136,14 +145,55 @@ def show_hw_malfunction_notification(self):
+                        replay=True,
+                    )
+                )
+            elif malfunction_id == "err_fan_not_spinning":
```
Please save these ids as constants, similar to the previous errors

JFSolution replied:
ohh yes I wanted to do it later and then I forgot

irlaec commented on octoprint_mrbeam/mrb_logger.py:
```diff
@@ -86,7 +85,12 @@ def log(self, level, msg, *args, **kwargs):
+                # If it's a byte string, convert it to Unicode using "utf-8" encoding
+                msg = unicode(msg, "utf-8")
         except TypeError:
             # If it's already unicode we get this TypeError
```
Now that we already check for unicode, maybe we could add a warn here to see if anything weird happens

JFSolution replied:
I don't see a benefit there, this is just to be able to log something that is already unicode

irlaec commented on octoprint_mrbeam/static/css/mrbeam.css:
```diff
@@ -1763,7 +1763,7 @@ ul.dropdown-menu li a {
 .ui-pnotify .alert-error a,
 .ui-pnotify .alert-danger a {
-  color: #b94a48;
+  color: #e25303;
```
I missed that :sweat_smile:

JFSolution replied:
It was not mentioned in the other ticket :)

irlaec commented on octoprint_mrbeam/static/js/app/view-models/feedback-widget.js:
```diff
@@ -62,13 +62,22 @@ $(function () {
             } else {
                 channel = "Stable";
             }
+            if (MRBEAM_LASER_HEAD_MODEL === "x") {
```
I've seen we have similar code in convert.js. Maybe it's a good idea to add the "x", "S", "0" in mrbeam.js so that we can access it from everywhere in the frontend. We already have something for the ids:
`mrbeam.laserheadModel = {
    S: 1,
    X: 3,
};`

irlaec commented on octoprint_mrbeam/static/js/app/view-models/user-notification.js:
```diff
@@ -1,44 +1,152 @@
+                            "If this doesn’t solve the problem, please %(opening_tag_support)sopen a support ticket%(closing_tag_support)s."
+                    ),
+                    self._getKowledgebaseArticleLinkSprintf(
+                        gettext("https://support.mr-beam.org/en") +
```
Is it enough with this gettext, or is it possible that the rest of the url is different too?

JFSolution replied:
The rest is the same, I can remove the text from them as the only thing interesting is the id

irlaec commented on octoprint_mrbeam/static/js/app/view-models/user-notification.js:
```diff
@@ -1,44 +1,152 @@
+                            "\n" +
+                            "Please %(opening_tag_support)sopen a support ticket%(closing_tag_support)s."
+                    ),
+                    self._getKowledgebaseArticleLinkSprintf(self.err)
```
I think we need to pass an empty url, before the error code
```suggestion
                    self._getKowledgebaseArticleLinkSprintf(None,self.err)
```

JFSolution replied:
I removed the error as this not needed anymore, I also renamed the function

Commit:
added errorcodes definition file
added PR feedback
added i2c missing device error
added unknwo error message to easily see missing

Commit:
add unittests
